pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -294,11 +294,14 @@ type Controller struct {
 	// taints instead of evicting Pods itself.
 	useTaintBasedEvictions bool
 
-	// if set to true, NodeController will taint Nodes based on its condition for 'NetworkUnavailable',
-	// 'MemoryPressure', 'PIDPressure' and 'DiskPressure'.
-	taintNodeByCondition bool
-
 	nodeUpdateQueue workqueue.Interface
+
+	// if set to true, Controller will skip taint nodes by condition.
+	//
+	// NOTE: used in scheduler integration test.
+	// TODO(draveness): remove this after the scheduler integration test TestTaintBasedEvictions
+	// could pass with TaintNodesByCondition enabled.
+	SkipTaintNodeByCondition bool
```

Comment:
why is this needed? why isn't TestTaintBasedEvictions passing without this?

Reply:
The tests would fail if we enable TaintNodeByConditions https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/82703/pull-kubernetes-integration/1174931892722470913